### PR TITLE
Accept more paths to devices for install

### PR DESCRIPTION
**Pull request by Itxaka**
Accepts any of the paths that udev adds for ease of use and are laying
under /dev/disk/by-X

As those are all softlinks to the actual device, it should be faily easy
to just read the link and use the final target as that should always be
a full device path (/dev/vda for example)

Fixes: https://github.com/kairos-io/kairos/issues/1879

Signed-off-by: Itxaka <itxaka@kairos.io>

### CODE REVIEW

**Review by mauromorales — APPROVED** [review]

File: pkg/types/v1/config.go [hunk 1]
```diff
@@ -76,6 +78,22 @@ type InstallSpec struct {
 // Sanitize checks the consistency of the struct, returns error
 // if unsolvable inconsistencies are found
 func (i *InstallSpec) Sanitize() error {
+	// Accept that the target can be a /dev/disk/by-{label,uuid,path,etc..} and resolve it into a /dev/device
```
[the remark this answers is not in the document]
Comment (mauromorales) [reply pkg/types/v1/config.go:81]: yup not a bad idea

File: pkg/config/spec.go [hunk 2]
```diff
@@ -47,6 +47,27 @@ const (
 	TiB
 )
 
+// TODO: Move it to the sdk
+func resolveTarget(fs v1.FS, target string) (string, error) {
```
Comment (Itxaka) [inline pkg/config/spec.go:51]: good here to test it directly on kairos via branches, if it works will move it into the sdk as its a very generic method that can be used everywhere, including immucore

**Review by jimmykarily — APPROVED** [review]
Code wise it looks ok. Needs manual testing (or even better, automated one :) ).

**Comment by Itxaka** [comment]
Trying to test it here: https://github.com/kairos-io/kairos/actions/runs/10990339501

**Pull request merged**